replication-api/src/main/java/com/connexta/replication/queue/QueueBroker.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.queue;
+
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+
+/**
+ * A queue broker defines the point of entry for retrieving references to queues and for monitoring
+ * multiple queues at the same time for tasks to be processed.
+ */
+public interface QueueBroker {
```

Comment:
We will need to expose statistics about queues so we can collect metrics on them. Doesn't need to be done as part of this PR though.